devito/finite_differences/elementary.py

```diff
@@ -0,0 +1,805 @@
+import sympy
+
+from devito.finite_differences.differentiable import (DifferentiableFunction, diffify)
+
+
+class factorial(DifferentiableFunction, sympy.factorial):
+    __sympy_class__ = sympy.factorial
+    __new__ = DifferentiableFunction.__new__
```

Comment:
all these `__new__ = ...` in this file can probably be dropped. I know you took inspiration from what I did in `Add/Mul/...` in an old PR of my (current master), but in my current branch I've removed them and everything still works like a charm, as expected I'd say, because w/ no override the chosen `__new__` is the one of the first superclass.

Reply:
done